Cesium3DTiles/src/QuadtreeRasterMappedTo3DTile.cpp

```diff
@@ -0,0 +1,154 @@
+#include "Cesium3DTiles/QuadtreeRasterMappedTo3DTile.h"
+#include "Cesium3DTiles/IPrepareRendererResources.h"
+#include "Cesium3DTiles/RasterOverlay.h"
+#include "Cesium3DTiles/RasterOverlayTileProvider.h"
+#include "Cesium3DTiles/Tileset.h"
+#include "Cesium3DTiles/TilesetExternals.h"
+
+namespace Cesium3DTiles {
+
+QuadtreeRasterMappedTo3DTile::QuadtreeRasterMappedTo3DTile(
+    const CesiumUtility::IntrusivePointer<RasterOverlayTile>& pRasterTile,
+    const CesiumGeometry::Rectangle& textureCoordinateRectangle)
+    : RasterMappedTo3DTile(pRasterTile, textureCoordinateRectangle) {}
+
+RasterMappedTo3DTile::MoreDetailAvailable
+QuadtreeRasterMappedTo3DTile::update(Tile& tile) noexcept {
+  if (this->getState() == AttachmentState::Attached) {
```

Comment:
This method appears closely tied to a quadtree at first glance, but really the only reason it uses the quadtree is to be able to find the raster tile's parent, grandparent, etc. What if `RasterOverlayTile` simply had a pointer to its parent, the way that `Tile` does? Could we get rid of `QuadtreeRasterMappedTo3DTile`, make `RasterMappedTo3DTile` non-polymorphic, and no longer allocate it on the heap?